cli/npm/resolution/graph.rs

```diff
@@ -1629,40 +1753,55 @@ mod test {
       packages,
       vec![
         NpmResolutionPackage {
-          id: NpmPackageNodeId::from_serialized("package-a@1.0.0").unwrap(),
+          pkg_id: NpmPackageId::from_serialized(
+            "package-a@1.0.0_package-peer@4.0.0"
```

Comment:
These package identifiers always include the descendant peer dep resolutions now. This means lockfiles will have more information in them, but that's necessary in order to differentiate resolutions.